File: android/src/main/java/com/witsystem/top/flutterwitsystem/unlock/UnlockInfo.java
package com.witsystem.top.flutterwitsystem.unlock;


import java.util.List;

/**
 * 开锁的回调
 */
public interface UnlockInfo {

    /**
     * 成功的回调
     *
     * @param deviceId
     * @param code
     */
    void success(String deviceId, int code);

    /**
     * 失败的回调
     *
     * @param error
     * @param code
     */
    void fail(String error, int code);

    /**
     * 电量的回调
     *
     * @param battery
     */
    void battery(String deviceId, int battery);


    /**
     * 发现附近多个设备的时候回调，只有在开启附近设备才有可能回调
     * @param devices
     * @param code
     */
    void devices(List<String> devices, int code);


}

File: ios/Classes/unlock/UnlockInfo.swift
//
//  UnlockInfo.swift
//  flutterwitsystem
//
//  Created by yyjjls on 2020/7/3.
//开锁回调

import Foundation

protocol UnlockInfo {
    
     /**
        * 成功的回调
        *
        * @param deviceId
        * @param code
        */
    func success(deviceId:String, code:Int);

       /**
        * 失败的回调
        *
        * @param error
        * @param code
        */
    func fail(error:String, code:Int);

       /**
        * 电量的回调
        *
        * @param b
        */
    func battery(deviceId:String,b:Int);
    
    
    /**
     *  发现附近有多个设备的回调 只有转开启附近设备的时候才有可能回调¬
     */
    func dDevice(deviceId:String);
    
    
    
}
